docs(3rd-party): Add section for third party tools (#10749)
espressif/arduino-esp32: docs/en/third_party_tools.rst

#################
Third Party Tools
#################

Here you will find documentation pages for third party tools that can be used with the ESP32 Arduino Core.

.. warning::

>   Note that these tools are **not maintained by the ESP32 Arduino Core team**, so we cannot provide support or guarantee that they will work as expected.
>   Documentation for these tools should be provided and maintained by the community.

Author: ```suggestion
  These tools are **not maintained by the ESP32 Arduino Core team**, so we cannot provide support or guarantee that they will work as expected.
  Each tool documentation should be provided and maintained by the community.
```
